code/modules/client/preferences.dm

```diff
@@ -115,8 +115,8 @@ GLOBAL_LIST_EMPTY(preferences_datums)
 		if(load_character())
 			// SKYRAT EDIT START - Sanitizing languages
 			for(var/lang_path as anything in languages)
-				var/datum/language/language = GLOB.language_datum_instances[lang_path]
-				if(!language || language.secret)
+				var/datum/language/language = new lang_path()
+				if(!(language.type in subtypesof(/datum/language)) || language.secret)
```

Comment:
I would probably add a comment above this if() to say that this ensures that languages that don't exist in code anymore don't linger in your languages prefs. Just so we don't get back to this situation again in the future.